flocker/control/httpapi.py

```diff
@@ -134,6 +136,32 @@ def render_if_matches(self, request, **route_arguments):
     return render_if_matches
 
 
+@lru_cache(1)
+def _extract_containers_state(deployment_state):
+    """
+    Turn the deployment state into serializable simple types for every
+    container in the deployment state.
+
+    N.B.: Callers of this function should not mutate the returned value as this
```

Comment:
Yea +1000... let me do some profiling.

The thing is... we need to return something _deeply_ pyrsistent, and then call serialize() on it externally.

This sounds expensive in the general case.